File: Hospital_Management_Microservice/Hospital_Management_NET/Hospital_Management/Models/Specialization.cs
using System;
using System.Collections.Generic;

namespace Hospital_Management.Models;

public partial class Specialization
{
    public long Id { get; set; }

    public string? Name { get; set; }

    public byte[]? Specializationimage { get; set; }

    public virtual ICollection<Doctor> Doctors { get; set; } = new List<Doctor>();
}

File: Hospital_Management_Microservice/hospital management_springboot/src/main/java/com/app/Entity/Specialization.java
package com.app.Entity;

import java.util.List;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Lob;
import javax.persistence.OneToMany;

import lombok.Getter;
import lombok.Setter;

@Entity
@Getter
@Setter
public class Specialization {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;

	private String name; // Example: Cardiology, Neurology

	@Lob
	@Column(columnDefinition = "LONGBLOB") // To store image as BLOB
	private byte[] specializationimage;

	@OneToMany(mappedBy = "specialization", cascade = CascadeType.ALL)
	private List<Doctor> doctors;
	
}
